### PR TITLE
fix: aligning the interceptor metrics vars with the OTEL spec

### LINKED ISSUE #1031: OTEL Env vars don't respect spec
### Report

We have merged the support for OTEL metrics, but the envs used don't respect the spec:
https://opentelemetry.io/docs/specs/otel/protocol/exporter/

### Expected Behavior

We should respect the spec, we use `KEDA_HTTP_OTEL_HTTP_COLLECTOR_ENDPOINT` when it should be `OTEL_EXPORTER_OTLP_ENDPOINT` (and probably supported automatically under the hood), or `KEDA_HTTP_OTEL_HTTP_COLLECTOR_INSECURE` instead of `OTEL_EXPORTER_OTLP_INSECURE` and other cases

### PR DESCRIPTION
**Pull request by zorocloud**
_Provide a description of what has been changed_

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Any necessary documentation is added, such as:
  - [`README.md`](/README.md)
  - [The `docs/` directory](./docs)
  - [The docs repo](https://github.com/kedacore/keda-docs)

Fixes #

Aligning the interceptor proxy metrics config env vars with the OTEL spec. Resolves #1031

### CODE REVIEW

**Review by zroubalik — APPROVED** [review]

File: interceptor/config/metrics.go [hunk 1]
```diff
@@ -7,19 +7,19 @@ import (
...
+	OtelHTTPCollectorEndpoint string `envconfig:"OTEL_EXPORTER_OTLP_METRICS_ENDPOINT" default:"localhost:4318"`
 	// Sets the OTLP headers required by the otel exporter
-	OtelHTTPHeaders string `envconfig:"KEDA_HTTP_OTEL_HTTP_HEADERS" default:""`
+	OtelHTTPHeaders string `envconfig:"OTEL_EXPORTER_OTLP_METRICS_HEADERS" default:""`
 	// Set the connection to the otel HTTP collector endpoint to use HTTP rather than HTTPS
-	OtelHTTPCollectorInsecure bool `envconfig:"KEDA_HTTP_OTEL_HTTP_COLLECTOR_INSECURE" default:"false"`
+	OtelHTTPCollectorInsecure bool `envconfig:"OTEL_EXPORTER_OTLP_METRICS_INSECURE" default:"false"`
 	// Set the interval in seconds to export otel metrics to the configured collector endpoint
-	OtelMetricExportInterval int `envconfig:"KEDA_HTTP_OTEL_METRIC_EXPORT_INTERVAL" default:"30"`
+	OtelMetricExportInterval int `envconfig:"OTEL_EXPORTER_OTLP_METRICS_EXPORT_INTERVAL" default:"30"`
```
Comment (JorTurFer) [inline interceptor/config/metrics.go:22]: I think that we shouldn't handle these vars in our code and delegate them directly to the SDK. Based on docs, it's already managing them:
- http: https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp
- grpc: https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc

(100% optional for this PR) We probably have to include another extra var: `OTEL_EXPORTER_OTLP_PROTOCOL`, based on it enable the HTTP exporter or the gRPC exporter. WDYT?

Comment (zorocloud) [reply interceptor/config/metrics.go:22]: Ah got you. Yeah that's a fair callout and also means less code should be needed too tbf. Will push an update when I get some time 👍

Comment (zroubalik) [reply interceptor/config/metrics.go:22]: @zorocloud any update on this please?

Comment (zorocloud) [reply interceptor/config/metrics.go]: Latest update should delegate the env var management to the OTEL SDK now @JorTurFer @zroubalik.

> 100% optional for this PR) We probably have to include another extra var: OTEL_EXPORTER_OTLP_PROTOCOL, based on it enable the HTTP exporter or the gRPC exporter. WDYT?

Apologies @JorTurFer, thought I responded to this! Agree re adding in an option for the gRPC exporter too. I'd prefer to leave it out of this PR though and can raise it in a separate one.

Comment (JorTurFer) [reply interceptor/config/metrics.go]: Sorry for the slow response, I've been involved on a huge working peak :( 
yeah, a followup PR is more than enough, thanks for your effort! ❤️

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**